### PR TITLE
From tables to h5py

### PR DESCRIPTION
**Pull request by Garyfallidis**
Switching from pytables to h5py. H5py still works with older numpy versions. pytables needs newer versions. Therefore h5py is easier to depend on. Also it seems that h5py gives better access to low level functions. Finally, testing in travis should be much simpler without the pytables dependency as that api has been changing quite a bit.

### CODE REVIEW

**Comment by Garyfallidis** [comment]
Still a bit more work to do. But there is progress.

**Comment by Garyfallidis** [comment]
@arokem I have only one bot failing now. The one with export DEPENDS="". What happens when DEPENDS is empty?. It installs minimum requirements by looking at requirements.txt or info.py or nothing? Can you check my .travis.yml file and suggest solutions?

**Comment by arokem** [comment]
I think you might need to add a `pip install -r requirements.txt` in the
`install` block, for the requirements to be read from that file.

**Comment by skoudoro** [comment]
@Garyfallidis, In this bot, `INSTALL_TYPE="setup"` so I think it does not use `requirement.txt` but you should add a `SetupDependency` in setup.py like you can see in this link: https://github.com/nipy/dipy/blob/master/setup.py#L150.

**Comment by Garyfallidis** [comment]
Oh!! Good catch @skoudoro ! :) Let me fix this.

**Comment by Garyfallidis** [comment]
Done. Let's wait for travis. I am not sure if the parameter heavy should be False or True.  The default option is False (and used for nibabel too) so I kept that. If anyone has other suggestions. Let me know.

**Comment by Garyfallidis** [comment]
👍 we are game! Merge if you think it's good.

File: setup.py [hunk 1]
```diff
@@ -150,6 +150,9 @@
 SetupDependency('nibabel', info.NIBABEL_MIN_VERSION,
                 req_type='install_requires',
                 heavy=False).check_fill(extra_setuptools_args)
+SetupDependency('h5py', info.H5PY_MIN_VERSION,
```
Comment (arokem) [inline setup.py]: I don't understand this. Why are we not using the requirements file to set up requirements that don't make crazy loops (such as numpy) ?

File: doc/dependencies.rst [hunk 2]
```diff
@@ -7,9 +7,9 @@ Dependencies
 Depends on a few standard libraries: python_ (the core language), numpy_ (for
 numerical computation), scipy_ (for more specific mathematical operations),
 cython_ (for extra speed) and nibabel_ (for file formats; we require version 2.1
-or higher). Optionally, it can use python-vtk_ (for visualisation), pytables_
+or higher). Optionally, it can use python-vtk_ (for visualisation), h5py_
```
Comment (arokem) [inline doc/dependencies.rst]: Not an optional dependency, is it?

File: doc/dependencies.rst [hunk 3]
```diff
@@ -7,9 +7,9 @@ Dependencies
...
 (for handling large datasets), matplotlib_ (for scientific plotting), and
-ipython_ (for interaction with the code and its results). cvxopt_ (version
-1.1.7 or later) is required for some modules.
+ipython_ (for interaction with the code and its results). cvxpy is required for
```
Comment (arokem) [inline doc/dependencies.rst]: Good catch!

File: doc/dependencies.rst (hunk 2 above)
Comment (Garyfallidis) [reply doc/dependencies.rst]: Nope, corrected.

File: setup.py (hunk 1 above)
Comment (Garyfallidis) [reply setup.py]: I just went with what we have done for nibabel here. You may want to chat with @matthew-brett about this.  He most likely used this for a good reason.

File: doc/dependencies.rst (hunk 3 above)
Comment (Garyfallidis) [reply doc/dependencies.rst]: :)

**Comment by arokem** [comment]
OK - we can follow up on the dependencies setup later. Merging!

**Pull request merged**